internal/resolver/resolver.go

```diff
@@ -58,43 +58,54 @@ func (r *Resolver) Resolve(reqs []*chart.Dependency, repoNames map[string]string
 	locked := make([]*chart.Dependency, len(reqs))
 	missing := []string{}
 	for i, d := range reqs {
-		constraint, err := semver.NewConstraint(d.Version)
-		if err != nil {
-			return nil, errors.Wrapf(err, "dependency %q has an invalid version/constraint format", d.Name)
-		}
-
-		if d.Repository == "" {
-			// Local chart subfolder
-			if _, err := GetLocalPath(filepath.Join("charts", d.Name), r.chartpath); err != nil {
-				return nil, err
-			}
+		var constraint *semver.Constraints
 
-			locked[i] = &chart.Dependency{
-				Name:       d.Name,
-				Repository: "",
-				Version:    d.Version,
+		// If version is defined
+		if d.Version != "" {
+			var err error
+			constraint, err = semver.NewConstraint(d.Version)
+			if err != nil {
+				return nil, errors.Wrapf(err, "dependency %q has an invalid version/constraint format", d.Name)
 			}
-			continue
 		}
-		if strings.HasPrefix(d.Repository, "file://") {
 
-			chartpath, err := GetLocalPath(d.Repository, r.chartpath)
-			if err != nil {
-				return nil, err
+		// Local chart
+		if d.Repository == "" || strings.HasPrefix(d.Repository, "file://") {
+			var chartpath string
+
+			if d.Repository == "" {
+				// From charts subfolder
+				var err error
+				chartpath, err = GetLocalPath(filepath.Join("charts", d.Name), r.chartpath)
+				if err != nil {
+					return nil, err
+				}
+			} else {
+				// From file:// repository
+				var err error
+				chartpath, err = GetLocalPath(d.Repository, r.chartpath)
+				if err != nil {
+					return nil, err
+				}
 			}
 
```

Comment:
remove `else`

```yaml
chartpath, err = GetLocalPath(d.Repository, r.chartpath)
if err != nil {
    return nil, err
}
```

Reply:
I don't understand. Why would I remove the `else`?

Reply:
The else is needed so I run the block of commands only when the repository is set with `file://`.

Reply:
@felipecrs When there are break, continue, goto or return statements in the if structure, the common way to write go code is to omit the else part.

Reply:
@yxxhero I agree. But the `return` which is inside of the `if` statement

https://github.com/helm/helm/blob/3f7a288ed2eb94b95727214d372dc8b3bb4f4504/internal/resolver/resolver.go#L76-L82

is only called when an error happens, and not always.

If I remove the `else`, `chartpath` would get assigned twice if no repository is set, with the wrong value btw.

I am really a first-time contributor here, and in Golang at all, so I am really sorry if I am missing something.

Reply:
@felipecrs you are right, it is good.

Reply:
That's great then :D